sdk/src/short_vec.rs

```diff
@@ -246,4 +277,17 @@ mod tests {
         let s = serde_json::to_string(&vec).unwrap();
         assert_eq!(s, "[[3],0,1,2]");
     }
+
+    #[test]
+    fn test_decode_len_aliased_values() {
+        let one1 = [0x01];
+        let one2 = [0x81, 0x00];
```

Comment:
Can we make this an error instead?

Reply:
Is there a chance we've stored one anywhere that doing so would change the result?

Reply:
If there is, I think we can solve that problem separately. Generally, I think we should look to make the protocol stricter when we can. It makes it easier to reason about.

Reply:
Right.  I agree, but chose this route to avoid an unintentional hard fork.  The other route is certainly cleaner as well

Reply:
Yeah, good point. This change is so low level, one could use it to cause a fork most anywhere.

Reply:
Could, did :balance_scale: